docs/task_guide.md

```diff
@@ -200,7 +200,11 @@ def construct_requests(self, doc, ctx):
     """
     return ...
 ```
-If your task requires generating text you'll need to return a `rf.greedy_until` request otherwise an `rf.loglikelihood` across all labels in a classification tasks will do.
+#### What's a `Request`? What's a `doc`? 
+To reiterate, a `doc` is just a `Dict` object that contains information about a document from your corpus. It can contain things like a prompt, question type information, answers and anything else you think will be needed in order to assess your model for a given task. Keep in mind that the fields of this can be basically whatever you want (you can sort this out in `training_docs` \ `validation_docs` \ `test_docs` if you need to customise things - see above), just remember to be consistent with them throughout the rest of the `Task` you write up.
+A `Request` is an object that takes the text prompt you want to present to a model and computes one of a few different types of response. These are evaluated lazily (meaning, only when the result is actually needed). If your task requires generating text you'll need to return a `rf.greedy_until` request otherwise an `rf.loglikelihood` across all labels in a classification tasks will do. There is the potential to create other requests such as one that returns extractive spans - if you want to contribute it, get in touch!
```

Comment:
I think the extractive spans one is a bad example because it's not clear what it would mean for a GPT model to give extractive spans aside from just generating the spans to be extracted, which would just use greedy_until